### ISSUE TITLE
macOS brew support?

### ISSUE THREAD

**Issue by caiohamamura** [rstudio/r-system-requirements#81]
Could we add in the guidelines to also include MacOS brew support?

The package names from there are really straightforward:

```
brew install gdal
brew install gsl
brew install hdf5
...
```

**Comment by caiohamamura** [comment]
If there are nothing against it I would love to contribute

**Comment by schloerke** [comment]
This would also allow [`rstudio/shiny` to not have to need custom installation for mac](https://github.com/rstudio/shiny/blob/2792d65e403141df08c0d07b1df3b2e4adb714b5/.github/workflows/R-CMD-check.yaml#L73-L87).

```
      # https://stackoverflow.com/a/66568545/591574
      #> fatal error: 'X11/Intrinsic.h' file not found
      - name: Install Cairo macOS R devel dependency
        if: runner.os == 'macOS' && matrix.config.r == 'devel'
        run: |
          brew install libxt
      # xquartz and cairo are needed for Cairo package.
      # harfbuzz and fribidi are needed for textshaping package.
      - name: Mac systemdeps
        if: runner.os == 'macOS'
        run: |
          brew install --cask xquartz
          brew install cairo
          brew install harfbuzz fribidi
```

**Comment by schloerke** [comment]
Update of my latest installation script: https://github.com/rstudio/shinycoreci/blob/main/.github/actions/install-mac-deps/action.yaml

Having RSPM macOS support would be a big step forward in supporting operating systems outside of Linux

**Comment by glin** [comment]
Hi, sorry for the delay in response. Support for macOS system requirements and precompiled binary packages are on the roadmap for RSPM, but I can't say where right now. And although brew support could potentially be added to this repo, it would still require macOS support in RSPM to be fully useful (system requirements API support, binaries).

This feedback helps a lot though, so thank you, I've added this as additional votes to the feature request internally.

In case it helps, [R-hub's sysreqsdb](https://github.com/r-hub/sysreqsdb) API and client package currently provides brew package names for a lot of packages.

**Comment by dmurdoch** [comment]
One issue is that `brew` builds are not always 100% compatible with CRAN builds of R.  The CRAN page `https://mac.r-project.org/bin/` contains builds of a lot of libraries.

Perhaps this means that MacOS would actually need to be two platforms:  one with brew builds, one with CRAN builds.

**Comment by gaborcsardi** [comment]
I am not sure if it is worth supporting individual system packages on macOS. On GHA we can install all packages at once with https://github.com/r-hub/actions/tree/main/setup-r-sysreqs and I am planning on adding this to rig as well. The file that has everything is about 340MB on arm64, which is not too bad, I think.

**Comment by glin** [comment]
Yep, it's been a while since that last response in 2021, but Package Manager now has precompiled macOS binaries. We also don't use brew at all, but just install CRAN's macOS libraries. GHA support makes sense for that.

**Comment by dmurdoch** [comment]
@gaborcsardi :  thanks for the pointer.

**Comment by schloerke** [comment]
> but Package Manager now has precompiled macOS binaries

Looking at `terra` https://p3m.dev/client/#/repos/cran/packages/overview?search=terra&distribution=macos (a package that had previous trouble installing), Package Manager has a binary all the way back to R 4.1 .

I'll take a look at removing the custom installation for the Shiny repos. Thank you!
